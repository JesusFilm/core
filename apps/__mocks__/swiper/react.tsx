import React from 'react'

export const Swiper = ({ children, 'data-testid': dataTestId }) => (
  <div data-testid={dataTestId}>{children}</div>
)
export const SwiperSlide = ({ children, 'data-testid': dataTestId }) => (
  <div data-testid={dataTestId}>{children}</div>
)
export const SwiperClass = {}
