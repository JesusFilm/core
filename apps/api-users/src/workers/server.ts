import { Job, Queue, Worker } from 'bullmq'
import { Logger } from 'pino'

import { connection } from './lib/connection'
import { logger } from './lib/logger'

const ONE_HOUR = 3600
const ONE_DAY = 86_400

function run({
  service,
  queueName,
  jobName,
  repeat
}: {
  service: (job: Job, logger?: Logger) => Promise<void>
  queueName: string
  jobName: string
  repeat?: string
}): void {
  // eslint-disable-next-line no-new
  new Worker(queueName, job, {
    connection
  })

  async function job(job: Job): Promise<void> {
    if (job.name !== jobName) return

    const childLogger = logger.child({
      queue: queueName,
      jobId: job.id
    })

    childLogger.info(`started job: ${job.name}`)
    await service(job, childLogger)
    childLogger.info(`finished job: ${job.name}`)
  }

  logger.info({ queue: queueName }, 'waiting for jobs')
}

async function main(): Promise<void> {
  run(
    await import(
      /* webpackChunkName: "email" */
      './email'
    )
  )
}

// avoid running on test environment
if (process.env.NODE_ENV !== 'test') void main()
