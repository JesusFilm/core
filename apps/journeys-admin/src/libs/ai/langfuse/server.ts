import { Langfuse } from 'langfuse'
import { LangfuseExporter } from 'langfuse-vercel'

export const langfuseEnvironment =
  process.env.VERCEL_ENV ??
  process.env.DD_ENV ??
  process.env.NODE_ENV ??
  'development'

export const langfuseExporter = new LangfuseExporter({
  environment: langfuseEnvironment
})

export const langfuse = new Langfuse({
  environment: langfuseEnvironment
})
