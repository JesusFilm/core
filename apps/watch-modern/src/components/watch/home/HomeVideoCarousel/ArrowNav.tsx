import { Button } from '@/components/ui/button'

interface ArrowNavProps {
  onPrev: () => void
  onNext: () => void
}

/**
 * Left and right arrow navigation controls
 */
export function ArrowNav({ onPrev, onNext }: ArrowNavProps) {
  return (
    <>
      {/* Previous Arrow */}
      <Button
        variant="ghost"
        size="icon"
        className="absolute left-4 top-1/2 -translate-y-1/2 z-20 h-12 w-12 rounded-full bg-black/20 backdrop-blur-sm text-white hover:bg-black/40 transition-all duration-200 hover:scale-110 focus-ring"
        onClick={onPrev}
        aria-label="Go to previous video slide"
      >
        <svg
          className="h-6 w-6"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          aria-hidden="true"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            d="M15 19l-7-7 7-7"
          />
        </svg>
      </Button>

      {/* Next Arrow */}
      <Button
        variant="ghost"
        size="icon"
        className="absolute right-4 top-1/2 -translate-y-1/2 z-20 h-12 w-12 rounded-full bg-black/20 backdrop-blur-sm text-white hover:bg-black/40 transition-all duration-200 hover:scale-110 focus-ring"
        onClick={onNext}
        aria-label="Go to next video slide"
      >
        <svg
          className="h-6 w-6"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          aria-hidden="true"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            d="M9 5l7 7-7 7"
          />
        </svg>
      </Button>
    </>
  )
}
