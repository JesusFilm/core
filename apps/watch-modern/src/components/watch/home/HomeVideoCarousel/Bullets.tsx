interface BulletsProps {
  total: number
  activeIndex: number
  progress: number
  onBulletClick: (index: number) => void
}

/**
 * Bullet indicators with progress fill
 */
export function Bullets({
  total,
  activeIndex,
  progress,
  onBulletClick
}: BulletsProps) {
  // Calculate progress percentage (0-100) for horizontal fill
  const progressPercent = Math.min((progress / 15) * 100, 100)

  return (
    <div className="absolute bottom-8 left-1/2 -translate-x-1/2 z-20 flex items-center gap-6">
      {/* Mixed Progress Indicators */}
      <div className="flex items-center gap-3">
          {Array.from({ length: total }, (_, index) => {
            const isActive = index === activeIndex
            
            if (isActive) {
              // Active slide: horizontal progress bar
              return (
                <button
                  key={index}
                  className="relative group transition-all duration-300 ease-out scale-100"
                  onClick={() => onBulletClick(index)}
                  aria-label={`Go to slide ${index + 1} of ${total}, ${Math.floor(progress)} seconds elapsed`}
                  aria-current="true"
                  style={{ 
                    width: '54px', 
                    height: '8px',
                    borderRadius: '4px'
                  }}
                >
                  {/* Background bar */}
                  <div 
                    className="absolute inset-0 rounded-full bg-white/30 transition-all duration-300"
                    data-testid={`progress-bar-bg-${index}`}
                  />
                  
                  {/* Progress fill */}
                  <div 
                    className="absolute top-0 left-0 h-full bg-white rounded-full transition-all duration-75 ease-linear"
                    style={{ 
                      width: `${progressPercent}%`,
                      boxShadow: '0 0 6px rgba(255,255,255,0.7)',
                      minWidth: progressPercent > 0 ? '8px' : '0px'
                    }}
                    data-testid={`progress-bar-fill-${index}`}
                  />
                </button>
              )
            } else {
              // Inactive slides: plain circles
              return (
                <button
                  key={index}
                  className={`relative group transition-all duration-300 ease-out ${
                    'opacity-60 hover:opacity-80 scale-90 hover:scale-100'
                  }`}
                  onClick={() => onBulletClick(index)}
                  aria-label={`Go to slide ${index + 1} of ${total}`}
                  aria-current="false"
                  style={{ 
                    width: '10px', 
                    height: '10px'
                  }}
                >
                  {/* Plain circle */}
                  <div 
                    className={`absolute inset-0 rounded-full transition-all duration-300 ${
                      index < activeIndex 
                        ? 'bg-white/70' // Completed slides - brighter
                        : 'bg-white/40 group-hover:bg-white/60' // Future slides - dimmer
                    }`}
                    data-testid={`circle-indicator-${index}`}
                  />
                </button>
              )
            }
          })}
      </div>
    </div>
  )
}
