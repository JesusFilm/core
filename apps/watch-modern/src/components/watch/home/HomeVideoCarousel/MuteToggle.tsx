import { Volume2, VolumeX } from 'lucide-react'

interface MuteToggleProps {
  isMuted: boolean
  onToggle: () => void
}

/**
 * Mute/unmute toggle button in bottom-right corner
 */
export function MuteToggle({ isMuted, onToggle }: MuteToggleProps) {
  return (
    <button
      className="absolute bottom-8 right-8 z-20 p-3 rounded-full bg-black/50 text-white ml-4 -mb-3 mr-1 transition-colors hover:bg-black/70"
      onClick={onToggle}
      aria-label={isMuted ? 'Unmute video carousel audio' : 'Mute video carousel audio'}
      aria-pressed={!isMuted}
    >
      {isMuted ? (
        <VolumeX className="h-6 w-6" />
      ) : (
        <Volume2 className="h-6 w-6" />
      )}
    </button>
  )
}
