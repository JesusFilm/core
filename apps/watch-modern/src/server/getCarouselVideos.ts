/**
 * Server-side data assembler for home video carousel.
 * Fetches video data from GraphQL Gateway API for the curated carousel.
 */

import { fetchGraphql } from '@/lib/graphql/fetchGraphql'
import { getLanguageIdFromLocale } from '@/lib/i18n/getLanguageIdFromLocale'
import { errorTelemetry } from '@/lib/telemetry'

// Types based on legacy VideoContentFields but simplified for carousel needs
export interface CarouselVideoItem {
  id: string
  label: string
  title: string
  description: string
  slug: string
  variantSlug: string | null
  hlsUrl: string | null
  imageUrl: string | null
  variantLanguagesCount: number
  languageSlugOverride: string | null
}

export interface CarouselSlug {
  slug: string
  languageSlugOverride?: string | null
}

// GraphQL query for individual carousel video - simplified version of legacy VideoContentFields
const GET_CAROUSEL_VIDEO = `
  query GetCarouselVideo($slug: ID!, $languageId: ID) {
    video(id: $slug, idType: databaseId) {
      id
      label
      title(languageId: $languageId, primary: true) {
        value
      }
      description(languageId: $languageId, primary: true) {
        value
      }
      slug
      variant {
        slug
        hls
      }
      images(aspectRatio: banner) {
        mobileCinematicHigh
      }
      variantLanguagesCount
    }
  }
`

export interface CarouselVideoResponse {
  video: {
    id: string
    label: string
    title: Array<{
      value: string
    }>
    description: Array<{
      value: string
    }>
    slug: string
    variant: {
      slug: string
      hls: string | null
    } | null
    images: Array<{
      mobileCinematicHigh: string | null
    }>
    variantLanguagesCount: number
  } | null
}

/**
 * Fetches carousel video data for the given slugs and locale.
 * @param slugs - Array of curated video slugs
 * @param locale - User's locale for content localization
 * @param maxItems - Maximum number of items to return (default: 10)
 * @returns Array of carousel video items, tolerant to missing videos
 */
export async function getCarouselVideos(
  slugs: CarouselSlug[],
  locale: string,
  maxItems: number = 10
): Promise<CarouselVideoItem[]> {
  console.log('🎬 [DEBUG] getCarouselVideos called with:', {
    slugCount: slugs.length,
    locale,
    maxItems,
    slugs: slugs.map(s => s.slug)
  })

  try {
    const languageId = getLanguageIdFromLocale(locale)
    console.log('🎬 [DEBUG] Resolved languageId:', languageId)

    const carouselItems: CarouselVideoItem[] = []

    // Fetch videos individually to handle missing videos gracefully
    const slugsToProcess = slugs.slice(0, maxItems)
    console.log('🎬 [DEBUG] Processing', slugsToProcess.length, 'slugs:', slugsToProcess.map(s => s.slug))

    for (const slugConfig of slugsToProcess) {
      console.log('🎬 [DEBUG] Fetching video for slug:', slugConfig.slug)

      try {
        const response = await fetchGraphql<CarouselVideoResponse>(
          GET_CAROUSEL_VIDEO,
          {
            slug: slugConfig.slug,
            languageId,
          }
        )

        console.log('🎬 [DEBUG] GraphQL response for', slugConfig.slug, ':', {
          hasData: !!response.data,
          hasVideo: !!response.data?.video,
          videoId: response.data?.video?.id || 'none'
        })

        if (response.data?.video) {
          const video = response.data.video
          const carouselItem = {
            id: video.id,
            label: video.label,
            title: video.title?.[0]?.value || 'Untitled Video',
            description: video.description?.[0]?.value || 'No description available',
            slug: video.slug,
            variantSlug: video.variant?.slug || null,
            hlsUrl: video.variant?.hls || null,
            imageUrl: video.images?.[0]?.mobileCinematicHigh || null,
            variantLanguagesCount: video.variantLanguagesCount,
            languageSlugOverride: slugConfig.languageSlugOverride || null,
          }

          carouselItems.push(carouselItem)
          console.log('✅ [DEBUG] Successfully added video:', {
            slug: slugConfig.slug,
            id: video.id,
            title: carouselItem.title,
            hasHlsUrl: !!carouselItem.hlsUrl,
            hasImageUrl: !!carouselItem.imageUrl
          })
        } else {
          console.warn('⚠️ [DEBUG] Video not found for slug:', slugConfig.slug)
        }
      } catch (videoError) {
        console.error('❌ [DEBUG] Failed to fetch video for slug', slugConfig.slug, ':', videoError)
        // Continue with other videos instead of failing completely
      }
    }

    console.log('🎬 [DEBUG] Final result:', {
      totalSlugsProcessed: slugsToProcess.length,
      successfulVideos: carouselItems.length,
      videos: carouselItems.map(v => ({
        slug: v.slug,
        title: v.title,
        hasHlsUrl: !!v.hlsUrl
      }))
    })

    return carouselItems
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : 'Unknown error'
    console.error('💥 [DEBUG] Critical error in getCarouselVideos:', error)

    // Log GraphQL error for monitoring (server-side telemetry would be handled by monitoring systems)
    // In production, this would be sent to error monitoring services
    if (typeof window !== 'undefined') {
      errorTelemetry.graphqlError(errorMessage, 'GetCarouselVideo')
    }

    // Return empty array on error to prevent breaking the page
    return []
  }
}
