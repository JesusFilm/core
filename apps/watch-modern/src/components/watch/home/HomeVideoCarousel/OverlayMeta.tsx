import Link from 'next/link'
import { CarouselVideoItem } from '@/server/getCarouselVideos'
import { Button } from '@/components/ui/button'
import { Container } from '@/components/ui/container'
import { buildWatchNowUrl } from '@/lib/urlUtils'

interface OverlayMetaProps {
  video: CarouselVideoItem
  watchUrl: string
}

/**
 * Overlay metadata component showing video title, type, description, languages, and Watch Now button
 */
export function OverlayMeta({ video, watchUrl }: OverlayMetaProps) {
  // Map label to user-friendly type
  const getVideoType = (label: string) => {
    const typeMap: Record<string, string> = {
      'film': 'Film',
      'series': 'Series',
      'documentary': 'Documentary',
      'course': 'Course',
      'story': 'Story',
      'testimony': 'Testimony'
    }
    return typeMap[label.toLowerCase()] || label
  }

  const watchNowUrl = buildWatchNowUrl(watchUrl, video.slug, video.languageSlugOverride)

  return (
    <Container className="absolute inset-0 flex flex-col justify-end p-4 sm:p-6 md:p-8 lg:p-12">
      <div className="w-full sm:w-1/2 lg:w-1/2 space-y-2 sm:space-y-3">
        {/* Title */}
        <h1
          className="text-carousel-title font-bold text-white leading-tight"
          style={{
            fontSize: 'clamp(1.5rem, 4vw, 2.5rem)',
            lineHeight: '1.1'
          }}
        >
          {video.title || 'Untitled Video'}
        </h1>

        <div className="flex flex-col sm:flex-row sm:items-center gap-3 sm:gap-4 pt-1">

        {/* Video Type Badge */}
        <div
          className="inline-flex items-center rounded-full bg-white/10 backdrop-blur-sm px-3 py-1 text-carousel-meta font-medium text-white focus-ring"
          role="status"
          aria-label={`Video type: ${getVideoType(video.label)}`}
        >
          {getVideoType(video.label)}
        </div>
        {/* Languages Available */}
        <div
            className="text-carousel-meta text-white/70"
            aria-label={`${video.variantLanguagesCount} language${video.variantLanguagesCount !== 1 ? 's' : ''} available`}
          >
            Available in {video.variantLanguagesCount} language{video.variantLanguagesCount !== 1 ? 's' : ''}
          </div>
        </div>

        {/* Description with enhanced truncation */}
        <p
          className="text-carousel-description text-white/80 leading-relaxed"
          style={{
            fontSize: '1rem',
            lineHeight: '1.4',
            display: '-webkit-box',
            WebkitLineClamp: 3,
            WebkitBoxOrient: 'vertical',
            overflow: 'hidden'
          }}
          aria-label={`Video description: ${video.description || 'No description available'}`}
        >
          {video.description || 'No description available'}
        </p>

        {/* Languages and Watch Now */}
        <div className="flex flex-col sm:flex-row sm:items-center gap-3 sm:gap-4 pt-1">
          

          {/* Watch Now Button */}
          <Button
            asChild
            size="lg"
            className="bg-white text-black hover:bg-white/90 font-semibold px-6 sm:px-8 py-3 rounded-full transition-all duration-200 hover:scale-105 focus-ring pointer-events-auto"
          >
            <Link
              href={watchNowUrl}
              className="flex items-center gap-2"
              aria-label={`Watch ${video.title || 'this video'} now - opens in new tab`}
            >
              <svg
                className="w-5 h-5"
                fill="currentColor"
                viewBox="0 0 20 20"
                aria-hidden="true"
              >
                <path
                  fillRule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  clipRule="evenodd"
                />
              </svg>
              Watch Now
            </Link>
          </Button>
        </div>
      </div>
    </Container>
  )
}
