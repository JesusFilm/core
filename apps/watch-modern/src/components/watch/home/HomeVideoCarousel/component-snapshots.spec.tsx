import { render } from '@testing-library/react'
import { ArrowNav } from './ArrowNav'
import { Bullets } from './Bullets'
import { MuteToggle } from './MuteToggle'
import { OverlayMeta } from './OverlayMeta'
import { CarouselVideoItem } from '@/server/getCarouselVideos'

// Mock Next.js Link component for OverlayMeta
jest.mock('next/link', () => ({
  __esModule: true,
  default: ({ children, href, ...props }: any) => (
    <a href={href} {...props} data-testid="link">
      {children}
    </a>
  )
}))

const mockVideo: CarouselVideoItem = {
  id: 'test-video-1',
  label: 'film',
  title: 'Test Movie Title',
  description: 'This is a test description for the video content.',
  slug: 'test-movie-slug',
  variantSlug: 'main',
  hlsUrl: 'https://example.com/video.m3u8',
  imageUrl: 'https://example.com/image.jpg',
  variantLanguagesCount: 12,
  languageSlugOverride: null
}

const watchUrl = 'https://www.jesusfilm.org/watch'

describe('Component Snapshots', () => {
  it('ArrowNav matches snapshot', () => {
    const { container } = render(
      <ArrowNav onPrev={() => {}} onNext={() => {}} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Bullets matches snapshot', () => {
    const { container } = render(
      <Bullets
        total={2}
        activeIndex={0}
        progress={7.5} // Half of 15 seconds
        onBulletClick={() => {}}
        onPlayPause={() => {}}
        isPlaying={true}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('MuteToggle matches snapshot when muted', () => {
    const { container } = render(
      <MuteToggle isMuted={true} onToggle={() => {}} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('MuteToggle matches snapshot when unmuted', () => {
    const { container } = render(
      <MuteToggle isMuted={false} onToggle={() => {}} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('OverlayMeta matches snapshot', () => {
    const { container } = render(
      <OverlayMeta video={mockVideo} watchUrl={watchUrl} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
