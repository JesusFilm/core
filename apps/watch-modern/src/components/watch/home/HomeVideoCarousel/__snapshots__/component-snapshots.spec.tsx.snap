// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshots ArrowNav matches snapshot 1`] = `
<button
  aria-label="Previous video"
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 cursor-pointer [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground absolute left-4 top-1/2 -translate-y-1/2 z-20 h-12 w-12 rounded-full bg-black/20 backdrop-blur-sm text-white hover:bg-black/40 transition-all duration-200 hover:scale-110 focus:ring-2 focus:ring-white/50"
>
  <svg
    aria-hidden="true"
    class="h-6 w-6"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      d="M15 19l-7-7 7-7"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`Component Snapshots Bullets matches snapshot 1`] = `
<div
  class="absolute bottom-6 left-1/2 -translate-x-1/2 z-20 flex items-center gap-3"
>
  <div
    class="flex gap-2"
  >
    <button
      aria-current="true"
      aria-label="Go to slide 1 of 2"
      class="relative h-3 w-3 rounded-full transition-all duration-300 bg-white scale-125"
    >
      <svg
        aria-hidden="true"
        class="absolute inset-0 w-full h-full -rotate-90"
        data-testid="progress-ring-0"
        viewBox="0 0 36 36"
      >
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          fill="none"
          stroke="rgba(255,255,255,0.2)"
          stroke-dasharray="100, 100"
          stroke-width="3"
        />
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          fill="none"
          stroke="white"
          stroke-dasharray="50, 100"
          stroke-linecap="round"
          stroke-width="3"
        />
      </svg>
    </button>
    <button
      aria-current="false"
      aria-label="Go to slide 2 of 2"
      class="relative h-3 w-3 rounded-full transition-all duration-300 bg-white/40 hover:bg-white/60"
    />
  </div>
  <button
    aria-label="Pause carousel"
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 cursor-pointer [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground h-8 w-8 rounded-full bg-black/20 backdrop-blur-sm text-white hover:bg-black/40 transition-all duration-200 ml-4"
  >
    <svg
      aria-hidden="true"
      class="h-4 w-4"
      fill="currentColor"
      viewBox="0 0 20 20"
    >
      <path
        clip-rule="evenodd"
        d="M6 4a1 1 0 00-1 1v10a1 1 0 001 1h1a1 1 0 001-1V5a1 1 0 00-1-1H6zM12 4a1 1 0 00-1 1v10a1 1 0 001 1h1a1 1 0 001-1V5a1 1 0 00-1-1h-1z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`Component Snapshots MuteToggle matches snapshot when muted 1`] = `
<button
  aria-label="Unmute video"
  aria-pressed="false"
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 cursor-pointer [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground absolute bottom-6 right-6 z-20 h-10 w-10 rounded-full bg-black/20 backdrop-blur-sm text-white hover:bg-black/40 transition-all duration-200 hover:scale-110 focus:ring-2 focus:ring-white/50"
>
  <svg
    aria-hidden="true"
    class="h-5 w-5"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      clip-rule="evenodd"
      d="M5.586 15H4a1 1 0 01-1-1v-4a1 1 0 011-1h1.586l4.707-4.707C10.923 3.663 12 4.109 12 5v14c0 .891-1.077 1.337-1.707.707L5.586 15z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
    <path
      d="M17 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`Component Snapshots MuteToggle matches snapshot when unmuted 1`] = `
<button
  aria-label="Mute video"
  aria-pressed="true"
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 cursor-pointer [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground absolute bottom-6 right-6 z-20 h-10 w-10 rounded-full bg-black/20 backdrop-blur-sm text-white hover:bg-black/40 transition-all duration-200 hover:scale-110 focus:ring-2 focus:ring-white/50"
>
  <svg
    aria-hidden="true"
    class="h-5 w-5"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.536 8.464a5 5 0 010 7.072M18.364 5.636a9 9 0 010 12.728M5.586 15H4a1 1 0 01-1-1v-4a1 1 0 011-1h1.586l4.707-4.707C10.923 3.663 12 4.109 12 5v14c0 .891-1.077 1.337-1.707.707L5.586 15z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`Component Snapshots OverlayMeta matches snapshot 1`] = `
<div
  class="absolute inset-0 flex flex-col justify-end p-6 md:p-8 lg:p-12"
>
  <div
    class="max-w-2xl space-y-4"
  >
    <div
      class="inline-flex items-center rounded-full bg-white/10 backdrop-blur-sm px-3 py-1 text-sm font-medium text-white"
    >
      Film
    </div>
    <h1
      class="text-2xl md:text-3xl lg:text-4xl font-bold text-white leading-tight"
    >
      Test Movie Title
    </h1>
    <p
      class="text-lg md:text-xl text-white/90 leading-relaxed line-clamp-3"
    >
      This is a test description for the video content.
    </p>
    <div
      class="flex flex-col sm:flex-row sm:items-center gap-4 pt-2"
    >
      <div
        class="text-white/70 text-sm"
      >
        Available in 
        12
         language
        s
      </div>
      <a
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 cursor-pointer [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 shadow h-10 bg-white text-black hover:bg-white/90 font-semibold px-8 py-3 rounded-full transition-all duration-200 hover:scale-105 flex items-center gap-2"
        data-testid="link"
        href="https://www.jesusfilm.org/watch/test-movie-slug"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <path
            clip-rule="evenodd"
            d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
            fill-rule="evenodd"
          />
        </svg>
        Watch Now
      </a>
    </div>
  </div>
</div>
`;
